tpm2/seal_test.go

```diff
@@ -79,31 +83,38 @@ func TestSealKeyToTPM(t *testing.T) {
 	}
 
 	t.Run("Standard", func(t *testing.T) {
-		tpm := openTPMForTesting(t)
-		defer closeTPM(t, tpm)
+		tpm, _, closeTPM := provision(t)
+		defer closeTPM()
 		run(t, tpm, &KeyCreationParams{PCRProfile: getTestPCRProfile(), PCRPolicyCounterHandle: 0x01810000})
 	})
 
 	t.Run("DifferentPCRPolicyCounterHandle", func(t *testing.T) {
-		tpm := openTPMForTesting(t)
-		defer closeTPM(t, tpm)
+		tpm, _, closeTPM := provision(t)
+		defer closeTPM()
 		run(t, tpm, &KeyCreationParams{PCRProfile: getTestPCRProfile(), PCRPolicyCounterHandle: 0x0181fff0})
 	})
 
-	t.Run("SealAfterProvision", func(t *testing.T) {
+	t.Run("SealWithNewConnection", func(t *testing.T) {
 		// SealKeyToTPM behaves slightly different if called immediately after EnsureProvisioned with the same Connection
-		tpm := openTPMForTesting(t)
-		defer closeTPM(t, tpm)
-		if err := tpm.EnsureProvisioned(ProvisionModeFull, nil); err != nil {
-			t.Errorf("Failed to provision TPM for test: %v", err)
-		}
+		tpm, tcti, _ := provision(t)
+		defer func() {
+			if err := tpm.Close(); err != nil {
+				t.Errorf("close failed: %v", err)
+			}
+		}()
+
+		tpm, _, _ = tpm2test.NewTPMConnectionFromExistingT(t, tpm, tcti)
 		run(t, tpm, &KeyCreationParams{PCRProfile: getTestPCRProfile(), PCRPolicyCounterHandle: 0x01810000})
 	})
 
 	t.Run("MissingSRK", func(t *testing.T) {
 		// Ensure that calling SealKeyToTPM recreates the SRK with the standard template
-		tpm := openTPMForTesting(t)
-		defer closeTPM(t, tpm)
+		tpm, tcti, _ := provision(t)
+		defer func() {
```

Comment:
wouldn't the 3rd ignored return arg from provision be usable in the defer here?

Reply:
It could - it would have to look like this though because the connection created here gets closed by the subsequent call to tpm2test.NewTPMConnectionFromExistingT(), which returns another close callback:
```
defer func() { closeTPM() }()
```